File: Python/_0453_Minimum_Moves_to_Equal_Array_Elements/Solution.py
# coding=utf-8
# Date: 2021/6/4 10:20
from typing import List


# 执行用时：56 ms, 在所有 Python3 提交中击败了67.45%的用户
# 内存消耗：15.6 MB, 在所有 Python3 提交中击败了89.60%的用户
class Solution:
    def minMoves(self, nums: List[int]) -> int:
        return sum(nums) - min(nums) * len(nums)


if __name__ == "__main__":
    print(Solution().minMoves([1, 2, 3]))  # 3

File: Java/src/_0453_Minimum_Moves_to_Equal_Array_Elements/Solution.java
package _0453_Minimum_Moves_to_Equal_Array_Elements;

import java.util.Arrays;

/**
 * Date: 2021/6/4 10:18
 * 执行用时：10 ms, 在所有 Java 提交中击败了41.17%的用户
 * 内存消耗：38.5 MB, 在所有 Java 提交中击败了97.21%的用户
 */
public class Solution {

    public int minMoves(int[] nums) {
        return Arrays.stream(nums).sum() - Arrays.stream(nums).min().getAsInt() * nums.length;
    }

    public static void main(String[] args) {

        System.out.println((new Solution()).minMoves(new int[]{1, 2, 3}));  // 3
    }
}
